qa/sc_evm_bootstrap.py

```diff
@@ -0,0 +1,253 @@
+        finally:
+            assert_true(exception_occurs, "Using a wrong key should fail")
+
+
+        evm_address = http_wallet_createPrivateKeySec256k1(sc_node_1)
+        evm_hex_address = to_checksum_address(evm_address)
+        logging.info("pubkey = {}".format(evm_address))
+
+        evm_address_2 = http_wallet_createPrivateKeySec256k1(sc_node_1)
+        logging.info("pubkey = {}".format(evm_address_2))
+
+        # call a legacy wallet api
+        ret = http_wallet_allPublicKeys(sc_node_1)
+        logging.info(ret)
+
+        ft_amount_in_zen = Decimal("33.22")
+        ft_amount_in_zennies = convertZenToZennies(ft_amount_in_zen)
+        ft_amount_in_wei = convertZenniesToWei(ft_amount_in_zennies)
+
+        # transfer some fund from MC to SC using the evm address created before
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      self.nodes[0],
+                                      evm_address,
+                                      ft_amount_in_zen,
+                                      mc_return_address)
+
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      self.nodes[0],
+                                      evm_address_2,
+                                      ft_amount_in_zen,
+                                      mc_return_address)
+
+        #input("\n\t======> Enter any input to continue generating a new sc block...")
+        generate_next_blocks(sc_node_1, "first node", 1)
+        self.sc_sync_all()
+        sc_best_block = sc_node_1.block_best()["result"]
+        assert_equal(sc_best_block["height"], 2, "The best block has not the specified height.")
+        assert_equal(875000000, sc_best_block['block']['header']['baseFee'])
+        logging.info(sc_best_block)
+        logging.info(sc_node_1.rpc_eth_getBalance(evm_hex_address, "1"))
+
+        # Check account balance via rpc not using optional tag
+        assert_equal(ft_amount_in_wei, int(sc_node_1.rpc_eth_getBalance(evm_hex_address)['result'], 16))
+
+        '''
+        # TODO
+        # Make and activate forging stake for the SC node 1
+        stake_amount = 1.234
+        sc_create_forging_stake_mempool(sc_node, stake_amount)
+        self.sc_sync_all()  # Sync SC nodes mempools
+        # Generate SC block with ForgerStake creation TX
+        generate_next_block(sc_node, "first node")
+        '''
+
+        #input("\n\t======> Enter any input to continue generating blocks till next consensus epoch...")
+        # Generate SC block on SC node 1 for the next consensus epoch
+        generate_next_block(sc_node_1, "first node", force_switch_to_next_epoch=True)
+        self.sc_sync_all()
+
+        sc_best_block = sc_node_1.block_best()["result"]
+        logging.info(sc_best_block)
+
+        assert_equal(765625000, sc_best_block['block']['header']['baseFee'])
```

Comment:
These object selectors can be very fragile and at the same time hard to debug. If they are used on an API response and the response is not as expected we don't get an exception about the API response - but rather about some invalid access of "block" or "header" into some object somewhere. We could refactor these to a helper method that does it in a more defensive manner and throws a more descriptive exception in case something goes wrong.